### PR TITLE
Support slash command localization

### PR DESCRIPTION
**Pull request by taahamahdi**

### CODE REVIEW

**Comment by taahamahdi** [comment]
@hoehoetvhss When you get a chance, can you take out the relevant `separated by commas` for the changes in this commit: https://github.com/Brainicism/KMQ_Discord/pull/1518/commits/a146520a62ba3f7c01b7f904e3bd9caefd04e1ce

**Comment by taahamahdi** [comment]
Looks good, thanks! I've removed the whitespaces from the `name` fields because Discord expects these to be single words. Can you verify the command names still make sense in this commit? https://github.com/Brainicism/KMQ_Discord/pull/1518/commits/48fcec3c68ae577713de3b13b06462751de59e3b

**Comment by hoehoetvhss** [comment]
The current translation is very different from the past. That's why everyone is making corrections, but it takes time.

**Comment by hoehoetvhss** [comment]
> The current translation is very different from the past. That's why everyone is making corrections, but it takes time.

But it doesn't interfere with the use.

File: src/commands/admin/appcommands.ts [hunk 1]
```diff
@@ -95,10 +100,47 @@ export default class AppCommandsCommand implements BaseCommand {
...
-                    const commands = command.slashCommands();
+                    const commands =
+                        command.slashCommands() as Array<Eris.ChatInputApplicationCommandStructure>;
+
                     for (const cmd of commands) {
+                        cmd.name =
+                            cmd.name ??
+                            i18n.translate(
+                                LocaleType.EN,
+                                `command.${commandName}.help.name`
```
Comment (Brainicism) [inline src/commands/admin/appcommands.ts:115]: Can we throw an exception if the 'name' translation key is missing, but we're expecting it to exist? Or a test

**Review by Brainicism — CHANGES_REQUESTED** [review]

**Comment by hoehoetvhss** [comment]
Any parts that I do not understand or are difficult to translate will be left in English for later translation.

**Review by Brainicism — APPROVED** [review]

**Pull request merged**